drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -0,0 +1,652 @@
+
+	usr_result.SSID.length = ssid.length;
+	memcpy(usr_result.SSID.value, ssid.value, ssid.length);
+
+	if (whd_wifi_scan(airoc_if, WHD_SCAN_TYPE_ACTIVE, WHD_BSS_TYPE_ANY, NULL, NULL, NULL, NULL,
+			  scan_cb_search, &scan_result, &(usr_result)) != WHD_SUCCESS) {
+		LOG_ERR("Failed start scan");
+		ret = -EAGAIN;
+		goto error;
+	}
+
+	if (k_sem_take(&airoc_wifi_data.sema_scan, K_MSEC(12 * 1000)) != 0) {
+		whd_wifi_stop_scan(airoc_if);
+		ret = -EAGAIN;
+		goto error;
+	}
+
+	if (usr_result.security == 0) {
+		ret = -EAGAIN;
+		goto error;
+	}
+
+	/* Connect to the network */
+	if (whd_wifi_join(airoc_if, &usr_result.SSID, usr_result.security, params->psk,
+			  params->psk_length) != WHD_SUCCESS) {
+		LOG_ERR("Failed to connect with network");
+		ret = -EAGAIN;
+		goto error;
+	}
+
+error:
+	if (ret < 0) {
+		net_if_dormant_on(data->iface);
+	} else {
+		net_dhcpv4_start(data->iface);
+		net_if_dormant_off(data->iface);
+		data->is_sta_connected = 1;
+	}
+
+	wifi_mgmt_raise_connect_result_event(data->iface, ret);
+	k_sem_give(&data->sema_common);
+	return ret;
+}
+
+static int airoc_mgmt_disconnect(const struct device *dev)
+{
+	int ret = 0;
+	struct airoc_wifi_data *data = (struct airoc_wifi_data *)dev->data;
+
+	if (k_sem_take(&data->sema_common, K_MSEC(AIROC_WIFI_WAIT_SEMA_MS)) != 0) {
+		return -EAGAIN;
+	}
+
+	if (whd_wifi_leave(airoc_if) != WHD_SUCCESS) {
+		k_sem_give(&data->sema_common);
+		ret = -EAGAIN;
+	} else {
+		data->is_sta_connected = 0;
+		net_dhcpv4_stop(data->iface);
+		net_if_dormant_on(data->iface);
+	}
+
+	wifi_mgmt_raise_disconnect_result_event(data->iface, ret);
+	k_sem_give(&data->sema_common);
+
+	return ret;
+}
+
+static void *ap_link_events_handler(whd_interface_t ifp, const whd_event_header_t *event_header,
+				    const uint8_t *event_data, void *handler_user_data)
+{
+	struct airoc_wifi_event_t airoc_event = {
+		.is_ap_event = 1,
+		.event_type = event_header->event_type
+	};
+
+	k_msgq_put(&airoc_wifi_msgq, &airoc_event, K_FOREVER);
+	return NULL;
+}
+
+static int airoc_mgmt_ap_enable(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	struct airoc_wifi_data *data = (struct airoc_wifi_data *)dev->data;
+	whd_security_t security;
+	whd_ssid_t ssid = {0};
+	uint8_t channel = 0;
+
+	if (k_sem_take(&data->sema_common, K_MSEC(AIROC_WIFI_WAIT_SEMA_MS)) != 0) {
+		return -EAGAIN;
+	}
+
+	if (data->second_interface_init == 0) {
+		if (whd_add_secondary_interface(data->whd_drv, NULL, &airoc_ap_if) !=
+		    CY_RSLT_SUCCESS) {
+			LOG_ERR("Error Unable to bring up the whd secondary interface");
+			k_sem_give(&data->sema_common);
+			return -EAGAIN;
+		}
+		data->second_interface_init = 1;
+	}
+
+	if (data->is_ap_up) {
+		LOG_ERR("Already AP is on - first disconnect");
+		k_sem_give(&data->sema_common);
+		return -EAGAIN;
+	}
+
+	ssid.length = params->ssid_length;
+	memcpy((void *)ssid.value, (void *)params->ssid, ssid.length);
+
+	if (((params->channel > 0) && (params->channel < 12)) ||
```

Comment:
This if statement is using a variety of values to check against params->channel's value. We should have a comment explaining the reason for the chosen values and why we default to channel 1 if outside those ranges.

Reply:
added comment:

	/* make sure to set valid channels for 2G and 5G:
	 * - 2G channels from1 to 11,  
	 * - 5G channels from 36 to 165
	 */